routing/datasource.go

```diff
@@ -494,6 +494,11 @@ func receiveRouteMatcher(o Options, out chan<- *routeTable, quit <-chan struct{}
 		select {
 		case defs := <-updatesRelay:
 			o.Log.Info("route settings received")
+
+			for i := range o.PreProcessors {
```

Comment:
it would be nice to have a test checkes that they are applied

Reply:
done